src/Npgsql/NpgsqlDataAdapter.cs

```diff
@@ -137,6 +139,65 @@ public new NpgsqlCommand InsertCommand
             get => (NpgsqlCommand)base.InsertCommand;
             set => base.InsertCommand = value;
         }
+
+        internal async Task<int> FillAsync(DataTable dataTable)
```

Comment:
Is the purpose here to add async APIs to NpgsqlDataAdapter (as proposed in https://github.com/dotnet/runtime/issues/22109)? If so, nothing seems to be exposed publicly (and I'd prefer for that to be separated out to another PR).

I'm also not sure these FillAsync methods should be calling internal APIs on DbCommand, rather than the public ones already exposed.

Reply:
`NpgsqlSchema.GetSchemaAsync()` requires `NpgsqlDataAdapter.FillAsync()` to be properly implemented. What do we do then?
- Block this PR until a separated PR adds async APIs to NpgsqlDataAdapter ?
- Call synchronous variant `Fill()` meanwhile?

Reply:
Coming back to this, I think this code is pretty much fine as-is - as long as we don't expose a public FillAsync API (yet), it doesn't really matter how we internally load schema data.

It could be slightly cleaner to have your custom FillAsync implementation handle the sync case as well (and receive an async parameter as is standard in the library), this way both sync and async go through the same path. But I wouldn't say that's very important, as long as everything works.